### ISSUE TITLE
entity_span_loss_fct

### ISSUE THREAD

**Issue by toughhou** [ShannonAI/mrc-for-flat-nested-ner#7]
https://github.com/ShannonAI/mrc-for-flat-nested-ner/blob/46ac876a3b86c9bf1ffcb853dee5170584993fca/layers/loss_func.py#L44

Hi, Here found some issue when debug.
1. Where start_positions defined?
2. Is entity_span_loss_fct defined correctly?
Per my understanding, here need to compute the probability that each possible span, which built from the start_positions and end_positions combination, also the loss.

span_loss = span_loss_fct(start_logits.view(-1, 2), start_positions.view(-1)) + **span_loss_fct(start_logits.view(-1, 2), start_positions.view(-1))**

**Comment by stepperL** [comment]
Many thanks for raising the issue. 
I will update the module "loss_func.py" and add code comments before 26th April.

**Comment by ChrisZhangcx** [comment]
The issue here. Don't know how to compute the loss for span match procedure.

**Comment by toughhou** [comment]
try the green ones.















--

Yours sincerely


Tough HOU
Tel: (+86) 185-6629-6654                   Email: toughhou@126.com




在 2020-04-23 20:56:20，"Chengxi Zhang" <notifications@github.com> 写道：

Same issue here. Don't know how to compute the loss for span match procedure.
